src/reducers/ComparisonSlice.ts

```diff
@@ -33,6 +43,56 @@ const comparison = createSlice({
   },
 });
 
+export const formatDownloadData = (
+  data: Record<string, ResultObject[]>,
+): object[] => {
+  const groupNames = Object.keys(data);
+  const transformedGroups: object[] = [];
+
+  groupNames.forEach((groupName) => {
+    const groupedResults = data[groupName].reduce(
+      (grouped: GroupedResults, result: ResultObject) => {
+        if (!grouped[result.header_name]) {
+          grouped[result.header_name] = [];
+        }
+        grouped[result.header_name].push(result);
+        return grouped;
+      },
+      {},
+    );
+
+    const transformedGroupEntries = Object.keys(groupedResults).map(
+      (header_name) => ({
+        [`${header_name} ${truncateHash(
+          groupedResults[header_name][0].new_rev,
+        )}`]: groupedResults[header_name],
+      }),
+    );
+
+    transformedGroups.push(...transformedGroupEntries);
+  });
```

Comment:
I'm not a big fan of using `forEach` or `reduce` for complicated things, I think this makes things generally difficult to read. I tend to prefer simple loops.

In this case for example, I think we could group the first reduce + the second map into just one loop, and push the result directly to `transformedGroups` as a result.

But let's not change this here and now, if you want you can look at it in a separate PR.

Reply:
I agree, it does not look easy to read ,but let's leave it for a future PR.